lib/ansible/parsing/yaml/constructor.py

```diff
@@ -120,6 +121,9 @@ def construct_yaml_seq(self, node):
     def construct_yaml_unsafe(self, node):
         return wrap_var(self.construct_yaml_str(node))
 
+    def construct_undefined(self, node):
+        return Undefined()
```

Comment:
Should we use `AnsibleUndefined` here instead?  I know there is likely circular imports due to `AnsibleUndefined` existing in `lib/ansible/template/__init__.py`, but I've wanted to move it to `lib/ansible/template/undefined.py` to allow it to be imported more freely.

Reply:
works for me, will try refactoring soon

Reply:
+1 for using `AnsibleUndefined`.

Reply:
Yes, this should use `AnsibleUndefined` in order to get the recursive Undefined behavior introduced in that class.

Reply:
@bcoca - This is a much needed feature.  I'm hoping you are safe and well and can look at this again sometime soon?
-C